### ISSUE TITLE
Add the .NET add-on as default add-on

### ISSUE THREAD

**Issue by LalatenduMohanty** [minishift/minishift#1819]
### General information

  * Minishift version: v1.9.0
  * OS: Linux / macOS / Windows
  * Hypervisor: KVM / xhyve / Hyper-V / VirtualBox

### Expected
User should be able to run .net applications on top of OpenShift provisioned through Minishift without importing any imagestreams i.e. Minishift should have the default image streams for .net.

### Actual
I am not sure if it is possible now. So it is more of a question for Minishift. 

-----------------------

### Update

Now that we have https://github.com/minishift/minishift-addons/tree/master/add-ons/dotnet , we should add this as a default add-on

**Comment by praveenkumar** [comment]
@LalatenduMohanty do we already have somewhere .net image stream to consume, if yes then we can have an addon which will do the work. Also, does this should be part of by default experience?

https://github.com/openshift/openshift-ansible/tree/master/roles/openshift_examples/files/examples/v3.7/image-streams here we have dotnet image stream which need to be tested.

**Comment by LalatenduMohanty** [comment]
> @LalatenduMohanty do we already have somewhere .net image stream to consume, if yes then we can have an addon which will do the work. Also, does this should be part of by default experience?

Yes, to both the questions. I am not sure what is the official source of .net image stream yet. But will findout.

**Comment by gbraad** [comment]
.NET core images are available, both based on RHEL and CentOS. But it is
also easy to make one.

-- 

   Gerard Braad | http://gbraad.nl
   [ Doing Open Source Matters ]

**Comment by gbraad** [comment]
@LalatenduMohanty what is your next step here? I am already writing an article and sxreen recordong to explain how to do this...

**Comment by LalatenduMohanty** [comment]
@gbraad The idea is to find out how users can run .net applications on a valinall set up of Minishift without manual steps.

**Comment by Bob-Davis** [comment]
.NET Core container images are available here:
https://access.redhat.com/containers/#/search/dotnet 

We should strive to use the official RHEL images vs. builds available from Microsoft, since there are slight, but important differences.

**Comment by gbraad** [comment]
Will pick this up with @tanaka_733

**Comment by tanaka-takayoshi** [comment]
Hi @LalatenduMohanty ,

At least I tested before, .NET Core image streams should be imported manually for RHEL based images.

```
 $ oc create --namespace openshift -f https://raw.githubusercontent.com/redhat-developer/s2i-dotnetcore/master/dotnet_imagestreams.json

$ oc create --namespace openshift -f https://raw.githubusercontent.com/redhat-developer/s2i-dotnetcore/master/templates/dotnet-example.json 
$ oc create --namespace openshift -f https://raw.githubusercontent.com/redhat-developer/s2i-dotnetcore/master/templates/dotnet-runtime-example.json 
$ oc create --namespace openshift -f https://raw.githubusercontent.com/redhat-developer/s2i-dotnetcore/master/templates/dotnet-pgsql-persistent.json
```

The steps are described in Red Hat .NET Core product document.
https://access.redhat.com/documentation/en-us/net_core/2.0/html/getting_started_guide/gs_dotnet_on_openshift#install_imagestreams

If you hope these imagestreams and templates are enabled by default, it should be discussed with the minishift product team.

**Comment by gbraad** [comment]
> If you hope these imagestreams and templates are enabled by default, it should be discussed with the minishift product team.

First step we could do is to wrap this in an add-on

**Comment by gbraad** [comment]
Repo:
  * https://github.com/redhat-developer/s2i-dotnetcore

Example:
  * https://github.com/redhat-developer/s2i-dotnetcore-ex

Description:
  * https://access.redhat.com/documentation/en-us/net_core/2.0/html/getting_started_guide/gs_dotnet_on_openshift

**Comment by LalatenduMohanty** [comment]
@gbraad I am interested to keep the add-on for .NET image stream add-on as part of default add-ons in Minishift. @minishift/minishift-dev  @Bob-Davis WDYT?

**Comment by anjannath** [comment]
If .NET is that important +1 for keeping it as a Default addon.

**Comment by budhrg** [comment]
> I am interested to keep the add-on for .NET image stream add-on as part of default add-ons in Minishift.

+1

**Comment by praveenkumar** [comment]
Whatever we put in default addons, become our responsibility to update as per Openshift release for imagestream and templates. Only if this is a hard requirement from CDK side then only we put it as default otherwise I don't have an issue with its current location.

**Comment by LalatenduMohanty** [comment]
Closing the issue as .NET imagestream and the builder application is available in Origin v3.9.0 https://github.com/minishift/minishift/pull/2112#issuecomment-377914861
